### ISSUE TITLE
JSON Arrays are not considered as valid JSON

### ISSUE THREAD

**Issue by Alex-D** [imbo/behat-api-extension#34]
Supposed I have that in my Scenario:

```cucumber
And the response body contains:
      """
      [
        {
          "id": 1,
          "gameId": 1
        },
        {
          "id": 2,
          "gameId": 2
        }
      ]
      """
```

The Assertion that check it's a stdClass failed, because it's an array of stdClass (ApiContext line 547):

```php
Assertion::isInstanceOf(
    $contains,
    'stdClass',
    'The supplied parameter is not a valid JSON object.'
);
```

If I remove this check, it works like a charm.

Maybe it would better to use the Guzzle's `json_decode`?

```php
\GuzzleHttp\json_encode($contains);
```

which throw a catchable InvalidArgumentExcepotion.

Are you OK with that?

Thanks!

**Comment by christeredvartsen** [comment]
The `Then the response body contains: <PyStringNode>` step is a step specifically made for checking JSON objects, but by using a special syntax you can check arrays entries as well:

```gherkin
And the response body contains:
    """
    {
        "[0]": {
            "id": 1,
            "gameId": 1
        },
        "[1]": {
            "id": 2,
            "gameId": 2
        }
    }
    """
```

I just saw your PR now and will have a deep look at it and see if there are any other implications. I will get back to this later today.

**Comment by Alex-D** [comment]
Yep, but it's not user-friendly to write and to read :/

It works well with my little modification :)

**Comment by christeredvartsen** [comment]
Resolved by ba9cc350aee0aa4e0ebb29980d34e6eae90e796a.
